src/Core/FSpot.FileSystem/GLibDirectory.cs

```diff
@@ -75,6 +75,7 @@ public IEnumerable<SafeUri> Enumerate (SafeUri uri)
 					yield return uri.Append (fileInfo.Name);
 					fileInfo.Dispose ();
 				}
+				fileEnumerator.Close (null);
```

Comment:
The fact that fileEnumerator is in a using block should probably dispose of this correctly?